src/Courses/CourseInterfaces.tsx

```diff
@@ -162,15 +182,36 @@ export class NewCourseUnitObj extends UnitObject {
 
 const newProblemUniqueGen = uniqueGen();
 
+export interface StudentGradeInstance {
+    id: number;
+    webworkQuestionPath: string;
+    problemNumber: number;
+    randomSeed: number;
+    // This is a jsonb field so it could be any (from db)
+    // Submitted in workbook used any so I'm going to keep it consistent here
+    // If this is used for form data we will never know any info about what keys are available
+    // Might make sense to make this an unknown type since I don't think we will ever access the types
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    currentProblemState: any;
+    studentGradeId: number;
+    studentTopicAssessmentInfoId: number;
+    scoreForBestVersion: number; // the score from the highest-scoring exam submission
+    overallBestScore: number; // the best score on this problem alone
+    active: boolean;
+    bestIndividualAttemptId: number;
+    bestVersionAttemptId: number;
+}
+
 export class StudentGrade {
+    gradeInstances?: StudentGradeInstance[];
     overallBestScore: number = 0;
     effectiveScore: number = 0;
     bestScore: number = 0;
     numAttempts: number = 0;
     numLegalAttempts: number = 0;
     locked: boolean = false;
     currentProblemState?: unknown;
-    id?: number;
+    id: number = 0;
```

Comment:
No, we should not be setting default values
Yes the nil checks are annoying, but all that is happening here is that we are disabling the typescript checks that protect us

We had a bug where course wasn't loading correctly because the id was defaulting to 0 so even though the course didn't load it tried to make the api call and fail